src/promptflow/promptflow/executor/flow_validator.py

```diff
@@ -238,6 +239,46 @@ def convert_flow_inputs_for_node(flow: Flow, node: Node, inputs: Mapping[str, An
                     ) from e
         return updated_inputs
 
+    @staticmethod
+    def _validate_aggregation_inputs(aggregated_flow_inputs: Mapping[str, Any], aggregation_inputs: Mapping[str, Any]):
+        """Validate the aggregation inputs according to the flow inputs."""
+        for key, value in aggregated_flow_inputs.items():
+            if key in aggregation_inputs:
+                raise InvalidAggregationInput(
+                    message_format="The input for aggregation is incorrect. The input '{input_key}' appears in both "
+                    "aggregated flow input and aggregated reference input. "
+                    "Please remove one of them and try the operation again.",
+                    input_key=key,
+                )
+            if not isinstance(value, list):
+                raise InvalidAggregationInput(
+                    message_format="The input for aggregation is incorrect. "
+                    "The value for aggregated flow input '{input_key}' should be a list. "
+                    "Please adjust the input value in your input data.",
```

Comment:
Shall we tell the user the type of the input to help troubleshoot?

"The value for aggregated flow input '{input_key}' should be a list but got {actual_value_type}."